modules/500-dashboard/templates/ingress.yaml

```diff
@@ -10,27 +11,35 @@ metadata:
     nginx.ingress.kubernetes.io/affinity: cookie
     nginx.ingress.kubernetes.io/affinity-mode: persistent
     nginx.ingress.kubernetes.io/session-cookie-name: d8-dashboard-affinity
-{{- if and (ne (include "helm_lib_module_https_mode" .) "Disabled") .Values.dashboard.auth.externalAuthentication }}
+    nginx.ingress.kubernetes.io/proxy-buffer-size: 32k
+  {{- if and (ne (include "helm_lib_module_https_mode" .) "Disabled") .Values.dashboard.auth.externalAuthentication }}
     nginx.ingress.kubernetes.io/auth-signin: {{ .Values.dashboard.auth.externalAuthentication.authSignInURL | quote }}
     nginx.ingress.kubernetes.io/auth-url: {{ .Values.dashboard.auth.externalAuthentication.authURL | quote }}
- {{- if .Values.dashboard.auth.externalAuthentication.useBearerTokens }}
+    {{- if .Values.dashboard.auth.externalAuthentication.useBearerTokens }}
     nginx.ingress.kubernetes.io/auth-response-headers: "authorization"
- {{- end }}
-    nginx.ingress.kubernetes.io/proxy-buffer-size: 32k
-{{- else }}
-
+    {{- end }}
+  {{- else }}
     nginx.ingress.kubernetes.io/auth-type: basic
     nginx.ingress.kubernetes.io/auth-secret: basic-auth
     nginx.ingress.kubernetes.io/auth-realm: "Authentication Required"
+  {{- end }}
     nginx.ingress.kubernetes.io/configuration-snippet: |
       proxy_ssl_certificate /etc/nginx/ssl/client.crt;
       proxy_ssl_certificate_key /etc/nginx/ssl/client.key;
       proxy_ssl_protocols TLSv1.2;
       proxy_ssl_session_reuse on;
-{{- end }}
-{{- if .Values.dashboard.auth.whitelistSourceRanges }}
+  {{- if or (not .Values.dashboard.auth.externalAuthentication) (eq (include "helm_lib_module_https_mode" .) "Disabled") }}
+    {{- if and $secret (hasKey $secret "data") (hasKey $secret.data "token") (index $secret.data "token") }}
+      proxy_set_header Authorization "Bearer {{ index $secret.data "token" | b64dec }}";
```

Comment:
And why do we need to set bearer token in ingress

Reply:
It is needed for the variant without dex in the cluster.